### PR TITLE
Save transformed router payload only if error

### PR DESCRIPTION
**Pull request by chandumlg**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: admin/admin.go [hunk 1]
```diff
@@ -135,6 +135,43 @@ func (a Admin) HeapDump(path *string, reply *string) (err error) {
...
+	if err != nil {
+		pkgLogger.Info("StartCPUProfile threw error. Cpu profiling may already be running")
+		*reply = "Cpu profiling may already be running"
+	} else {
+		*reply = "Cpu profile is being written to " + *path
+	}
+	return nil
+}
+
+// StartCpuProfile starts writing cpu profile at given path using pprof
```
Comment (snarkychef) [inline admin/admin.go:161]: fix comment

File: admin/admin.go [hunk 2]
```diff
@@ -135,6 +135,43 @@ func (a Admin) HeapDump(path *string, reply *string) (err error) {
...
+		}
+	}()
+	f, err := os.OpenFile(*path, os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
+		return err
+	}
+	pkgLogger.Info("Starting cpu profile. Writing to ", *path)
+	err = pprof.StartCPUProfile(f)
+	if err != nil {
+		pkgLogger.Info("StartCPUProfile threw error. Cpu profiling may already be running")
```
Comment (snarkychef) [inline admin/admin.go:153]: Can we handle only the already running error?

File: gateway/gateway.go [hunk 3]
```diff
@@ -163,7 +163,7 @@ func (gateway *HandleT) initUserWebRequestWorkers() {
 			batchRequestQ:  make(chan *batchWebRequestT),
 			reponseQ:       make(chan map[uuid.UUID]string),
 			workerID:       i,
-			batchTimeStat:  gateway.stats.NewStat("gateway.batch_time", stats.TimerType),
+			batchTimeStat:  gateway.stats.NewStat(fmt.Sprintf("gateway.batch_time_%d", i), stats.TimerType),
```
Comment (snarkychef) [inline gateway/gateway.go:166]: Let's use  tagged stat with workedId tag?
Also why do we need workerId?

File: gateway/gateway.go [hunk 4]
```diff
@@ -1102,7 +1102,7 @@ func (gateway *HandleT) Setup(application app.Interface, backendConfig backendco
 
 	gateway.batchSizeStat = gateway.stats.NewStat("gateway.batch_size", stats.CountType)
 	gateway.requestSizeStat = gateway.stats.NewStat("gateway.request_size", stats.CountType)
```
Comment (snarkychef) [inline gateway/gateway.go:1104]: We decided to use timer stat for request size and batch size right? Not db writes?

File: gateway/webhook/configuration.go [hunk 5]
```diff
@@ -19,6 +19,8 @@ func loadConfig() {
 	maxTransformerProcess = config.GetInt("Gateway.webhook.maxTransformerProcess", 64)
 	// Max time till when retries to source transformer are done
 	webhookRetryWaitMax = (config.GetDuration("Gateway.webhook.maxRetryTimeInS", time.Duration(10)) * time.Second)
+	// Min time gap when retries to source transformer are done
+	webhookRetryWaitMin = (config.GetDuration("Gateway.webhook.minRetryTimeInS", time.Duration(100)) * time.Millisecond)
```
Comment (snarkychef) [inline gateway/webhook/configuration.go:23]: This should be ending with `InMS` ?

File: gateway/webhook/webhookTransformer.go [hunk 6]
```diff
@@ -46,7 +46,9 @@ func (bt *batchWebhookTransformerT) transform(events [][]byte, sourceType string
 	if err != nil {
 		pkgLogger.Error(err)
 		bt.stats.failedStat.Count(len(events))
```
Comment (snarkychef) [inline gateway/webhook/webhookTransformer.go:48]: We don't need this as we are crashing anyways.

File: gateway/webhook/webhookTransformer.go [hunk 7]
```diff
@@ -46,7 +46,9 @@ func (bt *batchWebhookTransformerT) transform(events [][]byte, sourceType string
...
-		return transformerBatchResponseT{batchError: errors.New("Internal server error in source transformer")}
+		errorLog := fmt.Errorf("JS HTTP connection error to source transformer: URL: %v Error: %+v", url, err)
+		pkgLogger.Error(errorLog)
```
Comment (snarkychef) [inline gateway/webhook/webhookTransformer.go:50]: We don't need this as we are crashing anyways.

File: admin/admin.go (hunk 1 above)
Comment (chandumlg) [reply admin/admin.go:161]: done

File: admin/admin.go (hunk 2 above)
Comment (chandumlg) [reply admin/admin.go:153]: done

File: gateway/gateway.go (hunk 3 above)
Comment (chandumlg) [reply gateway/gateway.go:166]: done

File: gateway/gateway.go (hunk 4 above)
Comment (chandumlg) [reply gateway/gateway.go:1104]: done

File: gateway/webhook/configuration.go (hunk 5 above)
Comment (chandumlg) [reply gateway/webhook/configuration.go:23]: done

**Review by psrikanth88 — APPROVED** [review]

**Review by snarkychef — APPROVED** [review]

**Pull request merged**